pkg/agent/svid/rotator_test.go

```diff
@@ -149,7 +148,13 @@ func (s *RotatorTestSuite) TestRunWithUpdates() {
 	}
 
 	cancel()
-	s.Require().Equal(context.Canceled, t.Wait())
+	err = t.Wait()
```

Comment:
It might be slightly safer to add some timeout around this in case it takes a while for the goroutine(s) managed by the `tomb.Tomb` to safely close.
```suggestion
      select {
      case <-time.After(time.Second):
          s.T().Error("timed out while waiting for rotator goroutines to complete execution")
      case <-t.Dead():
          err = t.Err()
      }
```

Reply:
I feel like this adds more indeterminism to the test since success/failure now depends on how long it takes for the spawned goroutines to terminate which can vary depending on the conditions of the test system.  It might make more sense to define and enforce test timeouts globally instead of per-test.

Reply:
I think the thing we want to try to avoid here is a case where the test hangs indefinitely because the spawned goroutines are not closing properly, e.g. aren't respecting `ctx.Done()`.

In this case, I think it is preferable to let the test fail if the goroutines don't complete after some reasonable period of time. We wouldn't expect the rotator to be doing any processing that would take multiple seconds with this test configuration, even on a slow machine. I think your point about flakiness is valid though...we can also be even more conservative here with a higher timeout like 5-10 seconds to ensure this doesn't cause any test flakiness.

We don't enforce any global timeouts today in our test build targets beyond the existing default of `go test` of 10 minutes. I think in general global test timeouts are pretty hard to define with high confidence, except when they are so long as to be almost useless.

Reply:
After looking through some other test code in the project, I've seen that other tests (including one in this file) are handling test timeouts inconsistently today. Given that testing guidelines are not very firmly defined in [CONTRIBUTING.md](https://github.com/spiffe/spire/blob/master/CONTRIBUTING.md), I think I am ok to move this change forward for now.

I've created #2105 to track future efforts where we can more effectively handle this problem at the project level. It isn't all that reasonable to require every contributor writing tests to reinvent the wheel on what kinds of timeouts are acceptable.